python/mxnet/ndarray/numpy/_op.py

```diff
@@ -2432,3 +2433,249 @@ def indices(dimensions, dtype=_np.int32, ctx=None):
     else:
         raise ValueError("The dimensions must be sequence of ints")
 # pylint: enable=redefined-outer-name
+
+
+@set_module('mxnet.ndarray.numpy')
+def hanning(M, dtype=None, ctx=None):
```

Comment:
I still prefer to use float64 as the default type in the function argument instead of having a checker at the beginning of the function. The reason is that the type of `dtype` would be `Union[None, str, numpy.dtype]` in the current implementation, which is inconsistent with the docstring saying that  `dtype : str or numpy.dtype`. Also, it makes the semantic ambiguous and should be avoided especially in Python 3.7. If the default value of `dtype` in the argument is `np.float64` or 'float64' then we can enforce its type to be `Union[str, numpy.dtype]` as documented.

Reply:
it's good, i think i should use float32 as the defaulet type,